Karma, Grunt, Bower & Angular new versions

enanox opened the pull request:
Hi, I would like to submit the new Karma, Grunt, Bower & AngularJS (v1.12) support for the project,

Feel free to contact me for any comments.

Best regards.

enanox committed:
Regenerating Yeoman work dir. Added newest AngularJS & Karma.

enanox committed:
Add last yeoman and inject ng-route

enanox committed:
Add last Karma, Grunt & Bower versions

robinboehm commented on examples/wordcloud/index.html:
```diff
@@ -29,11 +29,15 @@
+    <script src="../../app/bower_components/angular-resource/angular-resource.js"></script>
+    <script src="../../app/bower_components/angular-cookies/angular-cookies.js"></script>
+    <script src="../../app/bower_components/angular-sanitize/angular-sanitize.js"></script>
+    <script src="../../app/bower_components/angular-route/angular-route.js"></script>
```
We don't need all this modules

robinboehm commented on package.json:
```diff
@@ -1,31 +1,42 @@
-  "name": "angular-d3",
-  "version": "0.0.1",
+  "name": "d3",
+  "version": "0.0.0",
```
Why you rename the module?

enanox replied:
Sorry, I will roll back this change. I regenerated the whole project with Yeoman (to benefit from the current usage of the plugins)

robinboehm commented on Gruntfile.js:
```diff
@@ -1,163 +1,395 @@
-                    }
-                ]
-            }
+            post: {}
```
Is this necessary?

enanox replied:
Do you mean the `post: {}` property or the whole `usemin` options?

robinboehm commented on bower.json:
```diff
@@ -0,0 +1,17 @@
+    "angular-resource": "1.2.15",
+    "angular-cookies": "1.2.15",
+    "angular-sanitize": "1.2.15",
+    "angular-route": "1.2.15"
```
d3 is a dependency

enanox replied:
Isn't d3 adapted to a custom `provider` in the module?

robinboehm replied:
Yeah, you're right.
But I think this is a really bad architecture.
We should change this.

https://github.com/robinboehm/angular-d3-directives/pull/1

enanox replied:
Using this workaround I think we could even wrap other libraries related with D3 such as TopoJSON, Crossfilter.js and so [link](http://www.ng-newsletter.com/posts/d3-on-angular.html)

enanox replied:
Hello, I changed the bower.json package name to consolidate against the package.json name of the project and added d3 as a dependency. Do you have any other consideration I should take?

robinboehm commented on examples/wordcloud/scripts/controllers/main.js:
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-angular.module('examplesApp')
+angular.module('examplesApp',['ngRoute', 'd3'])
```
Why we need ngRoute?

robinboehm commented:
Hey @enanox , thx for your help! :)

I've added some comments ...

enanox commented:
Sure, thanks for your feedback.

enanox committed:
Rolling back autogenerated package changes and ngRoute mistake.

enanox committed:
Fix CI configuration

enanox committed:
Add Travis build message to README. Until npm 0.8 is updated for the CI
virtual machines, remove the support for 0.8 builds. Delete unused post
property from Gruntfile.js

enanox committed:
Add d3 as a Bower dependency.

enanox committed:
Fix travis error for 0.8

enanox committed:
Restore core app load
